package.json

```diff
@@ -80,12 +82,9 @@
       }
     },
     "nonRenderableFiles": [
-      "sample.gif"
```

Comment:
Because it shows that only JS files are processed by react engine template and only non-JS files which are inside `nonRenderableFiles` array are copied to output folder.

Reply:
for me your sentence means that both `sample.gif` and `style.css` should be listed here

Reply:
`style.css` is used inside template for styling, but ok, I can left `sample.gif` as well. It's a only an example.

Reply:
wait 😄 let me explain why I ask

with nunjucks, if I do not put `sample.gif` in `nonRenderableFiles` then generation fails. How it looks like with react?

Reply:
Skip it :P

Reply:
so we basically do not need `nonRenderableFiles` in the current template 🤔 
is it mentioned already in the authoring document in generator? it should

we could maybe add some static js from some UI library or something, that does some magic and should be ignored? like code highlighter? or ...I don't know

Reply:
Incase you want to render a `.js` file as is (when it contains the File component as root) you would still need `nonRenderableFiles` 😄

Reply:
> is it mentioned already in the authoring document in generator? it should

It is stated [here](https://github.com/asyncapi/generator/blob/master/docs/authoring.md#common-assumptions-1) what the conditions for rendering a file is with React. I dont think it needs to be explained further should it? 🤔

Reply:
yeah, I'm affraid people jump on docs, do not read them 100% and now we have `nonRenderableFiles` described as

```
A list of file paths or globs that must be copied "as-is" to the target directory, i.e., without performing any rendering process. This is useful when you want to copy binary files.
```

> This is useful when you want to copy binary files. 

It is a clear reference to nunjucks and something not needed for react

Reply:
> It is a clear reference to nunjucks and something not needed for react

I have not tested this with the React renderer, I cant say if there is or if there isn't a special case where it can fail. So for me this is still a relevant option 😄 Also my comment regarding a `.js` file is still relevant.

Reply:
In React we only render `.js` files. If file is not `.js` and isn't in `nonRenderableFiles` array then generator skip it, if is in `nonRenderableFiles` copy it, including `.js` files which shouldn't be renderer :P

Reply:
🤦 right